avm/res/resources/deployment-script/main.bicep

```
@@ -195,7 +183,7 @@ resource deploymentScript 'Microsoft.Resources/deploymentScripts@2023-08-01' = {
     containerSettings: !empty(containerSettings) ? containerSettings : null
     storageAccountSettings: !empty(storageAccountResourceId) ? storageAccountSettings : null
     arguments: arguments
-    environmentVariables: !empty(environmentVariables) ? environmentVariables.secureList : []
+    environmentVariables: environmentVariables != null ? environmentVariables!.secureList : []
```

I think this could be simplified to

```suggestion
    environmentVariables: environmentVariables.?secureList ?? []
```
But no guarantees without testing 😄

---

I tried the coalesce way before, but didn't work for some reason.. Just tried it again and for some reason I got it working 😄. Maybe we should make it even simpler by removing `secureList`? There's no for it because of the secure decoration on the UDT.

---

Hey @johnlokerse,
if you can make it so that the `@secure` works with arrays, then I'd be all for it. It's non-support for array was the reason we used an object in the first place (in this and other modules)

---

Yes, that's true... forget about it. Will leave it as-is for now 😄 Auto-completion with the UDT makes it a lot better already.